stdlib/functools.pyi

```diff
@@ -52,25 +54,35 @@ if sys.version_info >= (3, 9):
         typed: bool
 
 @final
-class _lru_cache_wrapper(Generic[_T]):
-    __wrapped__: Callable[..., _T]
-    def __call__(self, *args: Hashable, **kwargs: Hashable) -> _T: ...
+class _lru_cache_wrapper(Generic[_P, _T]):
+    __wrapped__: Callable[_P, _T]
+    def __call__(self, *args: _P.args, **kwargs: _P.kwargs) -> _T: ...
     def cache_info(self) -> _CacheInfo: ...
     def cache_clear(self) -> None: ...
     if sys.version_info >= (3, 9):
         def cache_parameters(self) -> _CacheParameters: ...
 
-    def __copy__(self) -> _lru_cache_wrapper[_T]: ...
-    def __deepcopy__(self, __memo: Any) -> _lru_cache_wrapper[_T]: ...
+    def __copy__(self) -> _lru_cache_wrapper[_P, _T]: ...
+    def __deepcopy__(self, __memo: Any) -> _lru_cache_wrapper[_P, _T]: ...
+    if sys.version_info >= (3, 8):
+        @overload
+        def __get__(self, __instance: None, __owner: type[_S] | None = ...) -> _lru_cache_wrapper[_P, _T]: ...
+        @overload
+        def __get__(self, __instance: _S, __owner: type[_S] | None = ...) -> Callable[Concatenate[_S, _P], _T]: ...
+    else:
+        @overload
+        def __get__(self, __instance: None, __owner: type[_S] | None) -> _lru_cache_wrapper[_P, _T]: ...
+        @overload
+        def __get__(self, __instance: _S, __owner: type[_S] | None) -> Callable[Concatenate[_S, _P], _T]: ...
```

Comment:
```suggestion
    def __copy__(self) -> _lru_cache_wrapper[_F]: ...
    def __deepcopy__(self, __memo: Any) -> _lru_cache_wrapper[_F]: ...
    if sys.version_info >= (3, 8):
        def __get__(self, __instance: object | None, __owner: type[object] | None = ...) -> Self: ...
    else:
        def __get__(self, __instance: object | None, __owner: type[object] | None) -> Self: ...
```
I think this should always return itself, since you want the additional properties on the wrapper to be available regardless of whether you're accessing the wrapper from the instance or a class. It remaining callable with the same parameters is guaranteed by the `__call__` method.